### PR TITLE
[AMDGPU] Add pre-commit tests for PR90710. NFC.

### PR DESCRIPTION
**Pull request by dstutt**

### CODE REVIEW

**Comment by dstutt** [comment]
This highlights the issue as discussed in #90201

**Comment by jayfoad** [comment]
Can you just make this the first commit in #90710? Then reviewers can see the effect that the patch has on this test by looking at the two commits individually (but when you merge it it'll all get squashed into a single commit which is OK).

File: llvm/test/CodeGen/AMDGPU/image-sample-waw.mir [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py UTC_ARGS: --version 4
```
Comment (jayfoad) [inline llvm/test/CodeGen/AMDGPU/image-sample-waw.mir:1]: Maybe call it `waitcnt-sample-waw.mir` by analogy with the existing `waitcnt-vmem-waw.mir`?

**Comment by dstutt** [comment]
Committing this as part of #90710 instead

**Pull request closed without merging**